mlflow/models/python_api.py

```diff
@@ -19,6 +19,7 @@ def build_docker(
     name="mlflow-pyfunc",
     env_manager=_EnvManager.VIRTUALENV,
     mlflow_home=None,
+    install_java=False,
```

Comment:
Can we add a note block within the pydoc string to explain that this argument needs to be explicitly enabled to support container builds for SparkML and other JVM-dependent model flavors?

Reply:
Nope this doesn't need to be enabled for those flavors (we have a static list of flavors and always install java for them). This is mainly for custom pyfunc use case where we don't know users' model need java or not.

Reply:
Right, that's what I was getting at :) Custom PythonModel implementation where users are wrapping a SparkML implementation for deployment as a container on AML / Sagemaker. There are a number of organizations that do this and this is effectively a breaking change for their workflows. We'll just want to make it clear in the docs and the release notes that this functionality is changing.

Reply:
Ah makes sense, thanks for the explanation! Will add more docstring in this PR, and update docs accordingly in a follow-up task.